asapdiscovery-spectrum/asapdiscovery/spectrum/score_complex.py

```diff
@@ -0,0 +1,363 @@
+        docked_aligned = new_docking_dir / "aligned"
+        docked_aligned.mkdir(parents=True, exist_ok=True)
+
+        scorers = [ChemGauss4Scorer()]
+        # load addtional ml scorers
+        if inputs.ml_score:
+            from asapdiscovery.ml.models import ASAPMLModelRegistry
+            from asapdiscovery.docking.scorer import MLModelScorer
+            logging.info("Loading additional ML scorers")
+            # check which endpoints are availabe for the target
+            models = ASAPMLModelRegistry.reccomend_models_for_target(inputs.target)
+            ml_scorers = MLModelScorer.load_model_specs(models=models)
+            scorers.extend(ml_scorers)
+
+        # Prepare complex, re-dock and score
+        logging.info("Running protein prep, docking and scoring of %s", min_out)
+        scores_df, prepped_cmp, ligand_pose, aligned = dock_and_score(
+            min_out,
+            comp_name,
+            inputs.target,
+            scorers,
+            label=tag,
+            allow_clashes=True,
+            pdb_ref=file_ref,
+            aligned_folder=docked_aligned,
+            align_chain=inputs.dock_chain,
+            align_chain_ref=inputs.ref_chain,
+        )
+        logging.debug(
+            "Columns of scoring dataset from asapdiscovery: %s", scores_df.columns
+        )
+        scores_df["premin-score-POSIT"] = pre_min_score
+        df_save = scores_df[["premin-score-POSIT", "docking-score-POSIT"]]
+        if inputs.ml_score:  # Add ML scores  
+            if scores_df["docking-score-POSIT"].values:
+                df_save = scores_df[
+                    [
+                        "premin-score-POSIT",
+                        "docking-score-POSIT",
+                        "computed-SchNet-pIC50",
+                        "computed-E3NN-pIC50",
+                        "computed-GAT-pIC50",
+                    ]
+                ]
+            else:
+                df_save = pd.DataFrame(
+                    columns=[
+                        "premin-score-POSIT",
+                        "docking-score-POSIT",
+                        "computed-SchNet-pIC50",
+                        "computed-E3NN-pIC50",
+                        "computed-GAT-pIC50",
+                    ]
+                )
+
+        if ligand is not None:
+            df_save.insert(loc=0, column="lig-ID", value=ligand)
+        if prot_id is not None:
+            df_save.insert(loc=0, column="prot-ID", value=prot_id)
+        logging.debug("The DataFrame after docking is %s", df_save)
+
+        # Now save files for later scoring steps
+        docked_prepped = new_docking_dir / "prepped"
+        docked_prepped.mkdir(parents=True, exist_ok=True)
+        pdb_prep = docked_prepped / (aligned.stem + "_target.pdb")
+        sdf_ligand = docked_prepped / (aligned.stem + "_ligand.sdf")
+        # Really annoying, but Target and PreppedTarget have different functions for gen the PDB
```

Comment:
Non-blocking comment. Yes, this is annoying, we should probably raise an issue if this is something we should fix in our code.